sway-lsp/src/utils/keyword_docs.rs

```diff
@@ -0,0 +1,850 @@
+
+        let keywords = vec![
+            pub_keyword,
+            use_keyword,
+            as_keyword,
+            struct_keyword,
+            enum_keyword,
+            self_keyword,
+            fn_keyword,
+            trait_keyword,
+            impl_keyword,
+            for_keyword,
+            const_keyword,
+            return_keyword,
+            if_keyword,
+            else_keyword,
+            match_keyword,
+            mut_keyword,
+            let_keyword,
+            while_keyword,
+            where_keyword,
+            ref_keyword,
+            true_keyword,
+            false_keyword,
+            break_keyword,
+            continue_keyword,
+            str_keyword,
+            script_keyword,
+            contract_keyword,
+            predicate_keyword,
+            library_keyword,
+            dep_keyword,
+            abi_keyword,
+            storage_keyword,
+            asm_keyword,
+            deref_keyword,
+            configurable_keyword,
+        ];
+
+        keywords.iter().for_each(|keyword| {
+            let ident = keyword.ident.clone().to_string();
+            // remove "_keyword" suffix to get the keyword name
+            let name = ident.trim_end_matches("_keyword").to_owned();
+            let mut documentation = String::new();
+            keyword.attrs.iter().for_each(|attr| {
+                let tokens = attr.tokens.to_token_stream();
+                let lit = extract_lit(tokens);
+                writeln!(documentation, "{lit}").unwrap();
+            });
+            keyword_docs.insert(
+                name,
+                documentation.replace("///\n", "\n").replace("/// ", ""),
+            );
+        });
+
+        Self(keyword_docs)
+    }
+}
+
+impl std::ops::Deref for KeywordDocs {
+    type Target = HashMap<String, String>;
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+
+/// Extracts the literal from a token stream and returns it as a string.
+fn extract_lit(tokens: TokenStream) -> String {
+    let mut res = "".to_string();
+    for token in tokens.into_iter() {
+        if let TokenTree::Literal(l) = token {
+            let s = l.to_string();
+            let s = s.trim_start_matches("r\""); // remove the r" sequence at the beginning
+            let s = s.trim_end_matches('\"'); // remove the " at the end
+            res.push_str(s);
+        }
+    }
+    res
+}
+
+#[tokio::test]
+async fn keywords_in_sync() {
+    let keyword_docs = KeywordDocs::new();
+    let lsp_keywords: Vec<_> = keyword_docs.keys().collect();
+    let compiler_keywords: Vec<_> = sway_parse::RESERVED_KEYWORDS
+        .iter()
+        .map(|s| s.to_string())
+        .collect();
+
+    for keyword in &compiler_keywords {
+        if !lsp_keywords.contains(&keyword) {
+            let err =
+                format!("Error: Documention for the `{keyword}` keyword is not implemented in LSP");
+            panic!("{err}");
+        }
```

Comment:
nit: assert does the same thing

```suggestion
        assert!(
            lsp_keywords.contains(&keyword),
            "Error: Documention for the `{keyword}` keyword is not implemented in LSP"
        );
}
```

Reply:
nice, i'll add this change to the next PR